hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/DataEntity.java

```diff
@@ -147,29 +148,30 @@ static Role valueOf(String name) {
         }
 
         public Blueprint(DataRole role, StructuredData value, Map<String, Object> properties) {
-            super(role.name(), properties);
-            if (value == null) {
-                value = StructuredData.get().undefined();
-            }
-            this.role = role;
-            this.value = value;
+            this(role, value, properties, null, null);
         }
 
         public Blueprint(DataRole role, StructuredData value, Map<String, Object> properties,
                          Map<String, Set<CanonicalPath>> outgoing,
                          Map<String, Set<CanonicalPath>> incoming) {
             super(role.name(), properties, outgoing, incoming);
+            if (value == null) {
+                value = StructuredData.get().undefined();
+            }
             this.role = role;
```

Comment:
no. It should never be null, because it dubs for an ID. Basically, data entities cannot have an arbitrary ID but 1 of an enumerated list defined by the entity type that are being added under.